spec.html

```diff
@@ -4048,23 +4048,25 @@ <h1>Await</h1>
             1. Let _prevContext_ be the running execution context.
             1. Suspend _prevContext_.
             1. Push _asyncContext_ onto the execution context stack; _asyncContext_ is now the running execution context.
-            1. Resume the suspended evaluation of _asyncContext_ using NormalCompletion(_value_) as the result of the operation that suspended it.
+            1. Resume the suspended evaluation of _asyncContext_ with NormalCompletion(_value_).
             1. Assert: When we reach this step, _asyncContext_ has already been removed from the execution context stack and _prevContext_ is the currently running execution context.
             1. Return *undefined*.
           1. Let _onFulfilled_ be ! CreateBuiltinFunction(_fulfilledClosure_, 1, *""*, &laquo; &raquo;).
           1. Let _rejectedClosure_ be a new Abstract Closure with parameters (_reason_) that captures _asyncContext_ and performs the following steps when called:
             1. Let _prevContext_ be the running execution context.
             1. Suspend _prevContext_.
             1. Push _asyncContext_ onto the execution context stack; _asyncContext_ is now the running execution context.
-            1. Resume the suspended evaluation of _asyncContext_ using ThrowCompletion(_reason_) as the result of the operation that suspended it.
+            1. Resume the suspended evaluation of _asyncContext_ with ThrowCompletion(_reason_).
             1. Assert: When we reach this step, _asyncContext_ has already been removed from the execution context stack and _prevContext_ is the currently running execution context.
             1. Return *undefined*.
           1. Let _onRejected_ be ! CreateBuiltinFunction(_rejectedClosure_, 1, *""*, &laquo; &raquo;).
           1. Perform ! PerformPromiseThen(_promise_, _onFulfilled_, _onRejected_).
           1. Remove _asyncContext_ from the execution context stack and restore the execution context that is at the top of the execution context stack as the running execution context.
-          1. Set the code evaluation state of _asyncContext_ such that when evaluation is resumed with a Completion _completion_, the following steps of the algorithm that invoked Await will be performed, with _completion_ available.
-          1. Return.
-          1. NOTE: This returns to the evaluation of the operation that had most previously resumed evaluation of _asyncContext_.
+          1. Let _outerContext_ be the running execution context.
+          1. NOTE: _outerContext_ is the execution context for the evaluation of the operation that had most previously resumed evaluation of _asyncContext_.
```

Comment:
It's unclear what "most previously" is supposed to mean.

If it means "the longest ago", then that's wrong, because the execution context that first resumed evaluation of _asyncContext_ isn't even necessarily on the stack, let alone in that position on the stack.

And if it means "most recently", then that got a problem too. It's true that, in the status quo, the most recent `Resume the suspended evaluation of _asyncContext_` step must have come from the execution context that's below it on the stack. However, this PR introduces "resume" steps that go in the 'other direction', so if _asyncContext_'s function itself calls an Async/GeneratorFunction, then it may have been resumed several times since _outerContext_ resumed it.

So unless you want to introduce terminology that distinguishes the two 'directions' of resumes, I think it might be hard to describe how _outerContext_ relates to _asyncContext_ (other than being directly below it on the stack, of course).